pmd-core/src/main/java/net/sourceforge/pmd/PMDConfiguration.java

```diff
@@ -206,9 +206,15 @@ public ClassLoader getClassLoader() {
      * @param classLoader
      *            The ClassLoader to use
      */
+
+    private ClassLoader getClassLoaderFromClass(Class<?> clazz) {
+        return clazz.getClassLoader();
+    }
+
     public void setClassLoader(ClassLoader classLoader) {
         if (classLoader == null) {
-            this.classLoader = getClass().getClassLoader();
+            ClassLoader classLoader_1 = getClassLoaderFromClass(getClass());
```

Comment:
Line 216 and Line 217 can be combined since there is no change in Object Value assigned in Line 217. It is just a temporary storing of the value and is unnecessary using heap memory.
`this.classLoader = getClassLoaderFromClass(getClass());`